python/pyspark/pandas/tests/test_categorical.py

```diff
@@ -448,10 +451,7 @@ def identity(x) -> ps.Series[psdf.b.dtype]:  # type: ignore
         def astype(x) -> ps.Series[dtype]:
             return x.astype(dtype)
 
-        if LooseVersion(pd.__version__) >= LooseVersion("1.3"):
-            # TODO(SPARK-36367): Fix the behavior to follow pandas >= 1.3
-            pass
-        elif LooseVersion(pd.__version__) >= LooseVersion("1.2"):
+        if LooseVersion(pd.__version__) >= LooseVersion("1.2"):
```

Comment:
This behavior was change in pandas 1.3, right?
It seems to fail with pandas 1.2 now.
Also could you check with the older pandas?

Reply:
Thanks! Just fixed tests for pandas 1.0, 1.1, 1.2 series.